src/game/scenes/arena.c

```diff
@@ -1182,8 +1182,12 @@ void arena_dynamic_tick(scene *scene, int paused) {
         }
 
         // check some invariants
-        assert(obj_har[0]->pos.x >= ARENA_LEFT_WALL && obj_har[0]->pos.x <= ARENA_RIGHT_WALL);
-        assert(obj_har[1]->pos.x >= ARENA_LEFT_WALL && obj_har[1]->pos.x <= ARENA_RIGHT_WALL);
+        if(!player_frame_isset(obj_har[0], "ab")) {
+            assert(obj_har[0]->pos.x >= ARENA_LEFT_WALL && obj_har[0]->pos.x <= ARENA_RIGHT_WALL);
+        }
+        if(!player_frame_isset(obj_har[1], "ab")) {
+            assert(obj_har[1]->pos.x >= ARENA_LEFT_WALL && obj_har[1]->pos.x <= ARENA_RIGHT_WALL);
+        }
```

Comment:
```suggestion
        assert(player_frame_isset(obj_har[0], "ab") ||
               (obj_har[0]->pos.x >= ARENA_LEFT_WALL && obj_har[0]->pos.x <= ARENA_RIGHT_WALL));
        assert(player_frame_isset(obj_har[1], "ab") ||
               (obj_har[1]->pos.x >= ARENA_LEFT_WALL && obj_har[1]->pos.x <= ARENA_RIGHT_WALL));
```
Keep the conditionals inside the `assert` so the `player_frame_isset` calls are not present in release builds.